src/coreclr/pal/inc/pal_mstypes.h

```diff
@@ -271,35 +194,29 @@ typedef unsigned int *PUINT;
 typedef BYTE BOOLEAN;
 typedef BOOLEAN *PBOOLEAN;
 
-typedef unsigned __int8 UINT8;
-typedef signed __int8 INT8;
-typedef unsigned __int16 UINT16;
-typedef signed __int16 INT16;
-typedef unsigned __int32 UINT32, *PUINT32;
-typedef signed __int32 INT32, *PINT32;
-typedef unsigned __int64 UINT64, *PUINT64;
-typedef signed __int64 INT64, *PINT64;
-
-typedef unsigned __int32 ULONG32, *PULONG32;
-typedef signed __int32 LONG32, *PLONG32;
-typedef unsigned __int64 ULONG64;
-typedef signed __int64 LONG64;
-
-#if defined(HOST_X86) && _MSC_VER >= 1300
-#define _W64 __w64
-#else
-#define _W64
-#endif
+typedef uint8_t UINT8;
+typedef int8_t INT8;
+typedef uint16_t UINT16;
+typedef int16_t INT16;
+typedef uint32_t UINT32, *PUINT32;
+typedef int32_t INT32, *PINT32;
+typedef uint64_t UINT64, *PUINT64;
+typedef int64_t INT64, *PINT64;
+
+typedef uint32_t ULONG32, *PULONG32;
+typedef int32_t LONG32, *PLONG32;
+typedef uint64_t ULONG64;
+typedef int64_t LONG64;
 
 #ifdef HOST_64BIT
 
-#define _atoi64 (__int64)atoll
+#define _atoi64 (int64_t)atoll
 
-typedef __int64 INT_PTR, *PINT_PTR;
-typedef unsigned __int64 UINT_PTR, *PUINT_PTR;
-typedef __int64 LONG_PTR, *PLONG_PTR;
-typedef unsigned __int64 ULONG_PTR, *PULONG_PTR;
-typedef unsigned __int64 DWORD_PTR, *PDWORD_PTR;
+typedef int64_t INT_PTR, *PINT_PTR;
+typedef uint64_t UINT_PTR, *PUINT_PTR;
+typedef int64_t LONG_PTR, *PLONG_PTR;
+typedef uint64_t ULONG_PTR, *PULONG_PTR;
+typedef uint64_t DWORD_PTR, *PDWORD_PTR;
```

Comment:
Can these be changed to `intptr_t` and `uintptr_t` instead? This reduces ifdefs between 32bit and 64bit.

And also the `INTPTR_MAX` constant.

Reply:
They cannot. Changing them leads to build breaks, and fixing those build breaks would require introducing more complicated ifdefs elsewhere (or larger refactoring that I am not planning to do in this PR).